### PR TITLE
[Merged by Bors] - Use HDFS discovery configmap for deep storage

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- HDFS deep storage now configured via discover config map instead of url to namenode
- Added HDFS deep storage test
- Increased install-druid cluster timeouts in tests
- Adapted docs and examples

This is breaking due to a CRD change. Needs discussion?

closes https://github.com/stackabletech/druid-operator/issues/188

## Review Checklist

- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by sbernauer — COMMENTED** [review]
Not a full review as i currently did not find the time to review properly, sorry

File: examples/psql-s3/psql-s3-druid-cluster.yaml [hunk 1]
```diff
@@ -44,13 +56,26 @@ spec:
...
-    endpoint: s3-de-central.profitbricks.com
-    credentialsSecret: s3-credentials    # this needs to be deployed additionally
+  ingestion:
+    s3connection:
+      inline:
+        host: minio-druid
+        port: 9000
+        accessStyle: Path
+        credentials:
+          secretClass: druid-s3-credentials
```
[the remark this answers is not in the document]
Comment (maltesander) [reply examples/psql-s3/psql-s3-druid-cluster.yaml]: done

**Review by sbernauer — COMMENTED** [review]
Actual change LGTM. I will play around a little with the different CRD versions and see how that works

**Comment by sbernauer** [comment]
Current problem.
Maybe me kubectl apply is wrong as i have no idea what I'm doing ^^
```
sbernauer@sbernauer:~/stackabletech/druid-operator$ helm ls
NAME            NAMESPACE       REVISION        UPDATED                                         STATUS          CHART                   APP VERSION
druid-operator  default         1               2022-07-06 14:04:19.422411088 +0200 CEST        deployed        druid-operator-0.6.0    0.6.0      
sbernauer@sbernauer:~/stackabletech/druid-operator$ k get crd druidclusters.druid.stackable.tech -o yaml | grep alpha
  - name: v1alpha1
  - v1alpha1
sbernauer@sbernauer:~/stackabletech/druid-operator$ cargo r crd | kubectl apply -f -
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/stackable-druid-operator crd`
Warning: resource customresourcedefinitions/druidclusters.druid.stackable.tech is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
The CustomResourceDefinition "druidclusters.druid.stackable.tech" is invalid: status.storedVersions[0]: Invalid value: "v1alpha1": must appear in spec.versions
```

File: examples/simple-druid-cluster.yaml [hunk 2]
```diff
@@ -16,6 +16,44 @@ spec:
...
+    name: simple-zk
+---
+apiVersion: hdfs.stackable.tech/v1alpha1
+kind: HdfsCluster
+metadata:
+  name: simple-hdfs
+spec:
+  version: 3.2.2-stackable0.5.0
+  zookeeperConfigMapName: simple-hdfs-znode
+  dfsReplication: 3
```
[the remark this answers is not in the document]
Comment (maltesander) [reply examples/simple-druid-cluster.yaml]: done

**Comment by sbernauer** [comment]
Integration test run: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/10/console

**Review by sbernauer — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**